pkg/http/http.go

```diff
@@ -63,6 +80,106 @@ func (b BasicAuth) IsZero() bool {
 	return b.Username == "" && b.Password == "" && b.PasswordFile == ""
 }
 
+func NewClientConfigFromYAML(cfg []byte) (*ClientConfig, error) {
+	conf := &ClientConfig{}
+	if err := yaml.UnmarshalStrict(cfg, conf); err != nil {
+		return nil, err
+	}
+	return conf, nil
+}
+
+type httpClientOptions struct {
+	maxIdleConns        int
+	maxIdleConnsPerHost int
+}
+
+type HTTPClientOption func(options *httpClientOptions)
+
+func WithMaxIdleConns(maxIdleConns int) HTTPClientOption {
+	return func(opts *httpClientOptions) {
+		opts.maxIdleConns = maxIdleConns
+	}
+}
+
+func WithMaxIdleConnsPerHost(maxIdleConnsPerHost int) HTTPClientOption {
+	return func(opts *httpClientOptions) {
+		opts.maxIdleConnsPerHost = maxIdleConnsPerHost
+	}
+}
+
+// NewRoundTripperFromConfig returns a new HTTP RoundTripper configured for the
+// given http.HTTPClientConfig and http.HTTPClientOption.
+func NewRoundTripperFromConfig(cfg config_util.HTTPClientConfig, name string, optFuncs ...HTTPClientOption) (http.RoundTripper, error) {
+	opts := defaultHTTPClientOptions
+	for _, f := range optFuncs {
+		f(&opts)
+	}
+
+	newRT := func(tlsConfig *tls.Config) (http.RoundTripper, error) {
+		var rt http.RoundTripper = &http.Transport{
```

Comment:
This could be that I'm not looking correctly but aren't we missing certain options? Like `InsecureSkipVerify ` and `ResponseHeaderTimeout` ?

Reply:
> This could be that I'm not looking correctly but aren't we missing certain options? Like `InsecureSkipVerify ` and `ResponseHeaderTimeout` ?

ResponseHeaderTimeout yes, but AFAIR `InsecureSkipVerify` is handled by tls ?